lib/src/working_copy.rs

```diff
@@ -511,6 +559,64 @@ impl TreeState {
             })
             .collect();
 
+        let fsmonitor_clock_needs_save = fsmonitor_kind.is_some();
+        let changed_files: Option<Vec<PathBuf>> = match fsmonitor_kind {
+            None => None,
+            Some(FsmonitorKind::Test { changed_files }) => Some(changed_files),
+            #[cfg(feature = "watchman")]
+            Some(FsmonitorKind::Watchman) => match self.query_watchman(self.watchman_clock.clone())
+            {
+                Ok((watchman_clock, changed_files)) => {
+                    // TODO: is it okay to update the clock even if we return `Err` below?
```

Comment:
I would think that we might lose updates if we do

Reply:
Assigned at the end of the function. (Also, I made this transactional in https://github.com/martinvonz/jj/pull/1740 anyways.)